graveler/committed/merge_iterator.go

```diff
@@ -6,22 +6,31 @@ import (
 	"github.com/treeverse/lakefs/graveler"
 )
 
-type mergeIterator struct {
-	diffIt graveler.DiffIterator
-	val    *graveler.ValueRecord
-	base   Iterator
-	err    error
+type compareIterator struct {
+	errorOnConflict bool
+	diffIt          graveler.DiffIterator
+	val             *graveler.Diff
+	base            Iterator
+	err             error
+}
+
+type compareValueIterator struct {
+	*compareIterator
 }
 
 // NewMergeIterator accepts an iterator describing a diff from theirs to ours.
 // It returns a ValueIterator with the changes to perform on theirs, in order to merge ours into it,
 // relative to base as the merge base.
 // The iterator will return ErrConflictFound when it reaches a conflict.
-func NewMergeIterator(diffTheirsToOurs graveler.DiffIterator, base Iterator) (*mergeIterator, error) {
-	return &mergeIterator{diffIt: diffTheirsToOurs, base: base}, nil
+func NewMergeIterator(diffTheirsToOurs graveler.DiffIterator, base Iterator) *compareValueIterator {
+	return &compareValueIterator{compareIterator: &compareIterator{diffIt: diffTheirsToOurs, base: base, errorOnConflict: true}}
```

Comment:
so if `errOnConflict` is true, a merge operation will fail as soon as a single conflict is found. At that point, the calling user would probably want to resolve conflicts, and probably not do it one conflict at a time. How can a user get the full list of conflicting objects?

Reply:
Then you use the compare iterator.
I'll add better comments